src/Illuminate/Support/Arr.php

```diff
@@ -331,7 +331,17 @@ public static function isAssoc(array $array)
      */
     public static function only($array, $keys)
     {
-        return array_intersect_key($array, array_flip((array) $keys));
+        $attributeData = [];
+
+        foreach ((array) $keys as $key) {
+            if (! $keyValue = data_get($array, $key)) {
```

Comment:
Ahaaa, you mean `null`, `0`, `false`, etc..? You are right, this should be fixed.